datahub/data.py

```diff
@@ -51,10 +51,28 @@
     34,
 ]
 
+
+@pd.api.extensions.register_dataframe_accessor("opal")
+class OpalAccessor:
+    """Pandas custom accessor for appending new data to Opal dataframe."""
+
+    def __init__(self, pandas_obj: pd.DataFrame) -> None:
+        """Initialization of dataframe. Validation to be added."""
+        self._obj = pandas_obj
+
+    def append(self, data: dict[str, float]) -> None:
+        """Function to append new data to existing dataframe."""
+        concat_df = pd.concat([self._obj, append_opal_frame(data)])
+        self._obj = concat_df
```

Comment:
I should've realised this approach with `pd.concat` wouldn't work a couple of weeks ago. We have to update the df in-place. Pandas support extending the size of the DataFrame with `.loc` like this:

```suggestion
        row = get_opal_row(data)
        self._obj.loc[data["frame"]] = row  # type: ignore[call-overload]
```

Note: I am suggesting changing the name for the `append_opal_frame` function to `get_opal_row` because of changes I am suggestion be made to that function.